packages/compass-home/src/modules/telemetry.spec.ts

```diff
@@ -35,8 +39,10 @@ const dataService: Pick<DataService, 'instance' | 'getCurrentTopologyType'> = {
 
 describe('connection tracking', function () {
   let trackUsageStatistics: boolean;
+  let loggerAndTelemetry: LoggerAndTelemetry;
 
   before(async function () {
+    loggerAndTelemetry = createLoggerAndTelemetry('TEST-CONNECTION');
     // TODO(COMPASS-7397): Proper dependency injection for logger + telemetry would be nice here!
```

Comment:
You now have "dependency injection" added for those methods (through function params) and can actually fix this todo and detach tests from the implementation